### ISSUE TITLE
Using a different guard does not persist the authenticated user in the session. L5.8

### ISSUE THREAD

**Issue by istratos** [Adldap2/Adldap2-Laravel#750]
- Laravel Version: 5.8.26
- Adldap2-Laravel Version: 6.0.6
- PHP Version: 7.2.14
- LDAP Type: Active directory

### Description:
I will try to give the best explanation I can.

We, at our company, have several different Laravel Apps. One of them is a "master app" which manages the credentials for all our workers and what other apps they have access to and their roles/permissiong among them.

In our secondary apps, the login is performed using this package, with the connection defined in the User model to our main app, which works flawlessly!

However, we have a new app under development, which involves outside users (customers) to have access to a dashboard and do their stuff there.

Now this Auth is the main auth of the new app, since they also require account creation, password resets, email verify and whatnot, so we decided to make it the "main auth". (web guard and users provider)

In this case, our workers also have access to that app as an "admin", but they are still required to login via LDAP and ActiveDirectory using our main app instead of the Eloquent one which is used for our customers.

So I decided to create a secondary auth, a new LoginController and a new model for them, new guard and provider. I created the login form and coded everything so that it works.

Apparently everything is still working good since it checks for the credentials in our AD, it syncs them to DB and redirects to the /home after a successfull login, HOWEVER, the session is not being persisted. So we're redirected to the /home with a "successfull" login but actually there's no logged user!

This doesn't happen with the main auth obviously which is the Eloquent one, for our customers. That one works just fine.

If I use the "secondary" login (admins) using the `web` guard (default guard) but with `admin` provider instead of `users`, it works normally for the admin login. Obviously I can't share the same guard/provider for both authentications, so as soon as I switch to the `admin` guard, it stops working.

This is my config in the relevant files:

Both guards and providers in my `config/auth.php` file

```
'guards' => [
        'web' => [
            'driver' => 'session',
            'provider' => 'users',
        ],

        'admin' => [
            'driver' => 'session',
            'provider' => 'admins',
        ],
    ],

'providers' => [
        'users' => [
            'driver' => 'eloquent',
            'model' => App\User::class,
        ],

        'admins' => [
            'driver' => 'ldap', // was 'eloquent'
            'model' => App\Admin::class,
        ],
    ],
```

`ldap_auth.php`
```
'provider' => Adldap\Laravel\Auth\DatabaseUserProvider::class,
'model' => App\Admin::class,
```
And defining the admin guard in the login controller (this controller is just a copy paste from the original but with the other auth-related stuff

`AdminLoginController.php`
```
    protected function guard()
    {
        return Auth::guard('admin');
    }
```
`laravel.log`
```
[2019-07-01 09:28:23] local.INFO: User 'John Doe' has been successfully found for authentication.  
[2019-07-01 09:28:23] local.INFO: User 'John Doe' is being synchronized.  
[2019-07-01 09:28:23] local.INFO: User 'John Doe' has been successfully synchronized.  
[2019-07-01 09:28:23] local.INFO: User 'John Doe' is authenticating with username: 'CN=John Doe,OU=foo,DC=company,DC=bar'  
[2019-07-01 09:28:23] local.INFO: User 'John Doe' has successfully passed LDAP authentication.  
[2019-07-01 09:28:23] local.INFO: User 'John Doe' has been successfully logged in. 
```

TLDR;
When using a different guard other than the default in the login controller for a LDAP Auth, the authentication works just fine, but it doesn't persist the auth user into the session.

Thank you very much!

**Comment by istratos** [comment]
My mistake, everything was indeed working as intended. I was looking for the user with:
``` 
Auth::user();
```

Which returns null because of course, it's looking for it inside the default guard.

```
Auth::guard('admin')->user();
```
Returns my authenticated user just normally.

I'll close the issue and leave it here just in case someone else stumbles upon this issue. Thanks !
